package/src/plugins/add-integration.ts

```diff
@@ -0,0 +1,23 @@
+import { definePlugin } from "../core/define-plugin.js";
+import {
+	type AddIntegrationParams,
+	addIntegration,
+} from "../utilities/add-integration.js";
+
+export const addIntegrationPlugin = definePlugin({
+	name: "addIntegration",
+	hook: "astro:config:setup",
+	implementation:
+		({ updateConfig, config, logger }) =>
+		(
+			integration: Pick<AddIntegrationParams, "integration">["integration"],
+			options?: Pick<AddIntegrationParams, "options">["options"],
+		) =>
+			addIntegration({
+				integration,
+				options: options ?? {},
```

Comment:
```suggestion
				options,
```
I think that's enough?